[3.0 Beta] Wrong @payload-config path in create-payload-app

### Link to reproduction

_No response_

### Describe the Bug

When using `create-payload-app` on an existing nextjs application, the `tsconfig.json` is updated to add
```
 "@payload-config": [
        "./src/payload.config.ts"
]
```

to your `compilerOptions.paths`.
This seems to always assume a `"compilerOptions.baseUrl": "."` (Or none set, didn't check).

In my case, `baseUrl` was set to `src`, so the correct path would be:
```
"@payload-config": [
        "./payload.config.ts"
]
```

The setup script should try to get the path to `payload.config.ts` relative to your `baseUrl`.


### To Reproduce

1. Create a new nextjs application, set "compilerOptions.baseUrl": "src" in your `tsconfig.json`
2. Run `pnpm dlx create-payload-app@beta` (pnpm is what I used, probably same with npx)
3. Start dev mode, observe `Module not found: Can't resolve '@payload-config'`

### Payload Version

3.0.0-beta.6

### Adapters and Plugins

_No response_

--- a/packages/create-payload-app/src/lib/init-next.ts
+++ b/packages/create-payload-app/src/lib/init-next.ts
@@ -117,13 +117,17 @@ async function addPayloadConfigToTsConfig(projectDir: string, isSrcDir: boolean)
     warning(`Could not find tsconfig.json to add @payload-config path.`)
     return
   }
-
   const userTsConfigContent = await readFile(tsConfigPath, {
     encoding: 'utf8',
   })
   const userTsConfig = parse(userTsConfigContent) as {
     compilerOptions?: CompilerOptions
   }
+
+  const hasBaseUrl =
+    userTsConfig?.compilerOptions?.baseUrl && userTsConfig?.compilerOptions?.baseUrl !== '.'
+  const baseUrl = hasBaseUrl ? userTsConfig?.compilerOptions?.baseUrl : './'
+
   if (!userTsConfig.compilerOptions && !('extends' in userTsConfig)) {
     userTsConfig.compilerOptions = {}
   }
@@ -134,7 +138,7 @@ async function addPayloadConfigToTsConfig(projectDir: string, isSrcDir: boolean)
   ) {
     userTsConfig.compilerOptions.paths = {
       ...(userTsConfig.compilerOptions.paths || {}),
-      '@payload-config': [`./${isSrcDir ? 'src/' : ''}payload.config.ts`],
+      '@payload-config': [`${baseUrl}${isSrcDir ? 'src/' : ''}payload.config.ts`],
     }
     await writeFile(tsConfigPath, stringify(userTsConfig, null, 2), { encoding: 'utf8' })
   }

